integration-tests/src/test/java/com/datastax/oss/driver/mapper/NullSavingStrategyIT.java

```diff
@@ -66,14 +69,30 @@ public class NullSavingStrategyIT {
 
   @BeforeClass
   public static void setup() {
-    CqlSession session = SESSION_RULE.session();
-    session.execute(
+    // Create V3 protocol client for use in the tests
+    v3Session =
+        SessionUtils.newSession(
+            CCM_RULE,
+            SESSION_RULE.keyspace(),
+            DriverConfigLoader.programmaticBuilder()
+                .withString(DefaultDriverOption.PROTOCOL_VERSION, "V3")
+                .build());
+    v3Session.execute(
         SimpleStatement.builder(
                 "CREATE TABLE product_simple(id uuid PRIMARY KEY, description text)")
             .setExecutionProfile(SESSION_RULE.slowProfile())
             .build());
 
-    mapper = new NullSavingStrategyIT_InventoryMapperBuilder(session).build();
+    // Use V3 client in the InventoryMapper used in the tests
+    mapper = new NullSavingStrategyIT_InventoryMapperBuilder(v3Session).build();
+  }
```

Comment:
Per my comment above... doesn't this accomplish exactly the same thing without requiring a new field (or a teardown method)?

```java
  @BeforeClass
  public static void setup() {
    
    // JAVA-3076: Perform setup using a one-off v5 session to avoid any errors when using v3
    DriverConfigLoader setupSessionLoader = DriverConfigLoader.programmaticBuilder()
            .withString(DefaultDriverOption.PROTOCOL_VERSION, "V5")
            .build());
    try (CqlSession setupSession = SessionUtils.newSession(CCM_RULE, SESSION_RULE.keyspace(), setupSessionLoader)) {

      setupSession.execute(
              SimpleStatement.builder(
                              "CREATE TABLE product_simple(id uuid PRIMARY KEY, description text)")
                      .setExecutionProfile(SESSION_RULE.slowProfile())
                      .build());
    }

    // JAVA-3076: Actual test should continue to use v3 just like it always has
    mapper = new NullSavingStrategyIT_InventoryMapperBuilder(SESSION_RULE.session()).build();
  }
```

The code snippet above assumes that SESSION_RULE remains identical to what it current is in 4.x.

Reply:
The error isn't coming from the table create in the `setup()` call but the keyspace creation `SessionRule#before()` - which is invoked by junit as part of @ClassRule - so I don't the snippet fixes this issue. I'll rework the comments in this file to make that clearer